### ISSUE TITLE
TreeTable/DataTable:  sticky header does not display correctly embedded inside Splitter

### ISSUE THREAD

**Issue by joepascual** [primefaces/primefaces#11055]
### Describe the bug

My application uses the Splitter component to display two panels on the screen.  The left panel contains a TreeTable component and the right panel displays details of the selected row in the TreeTable.  When the TreeTable **_stickyHeader_** attribute is set to true, the TreeTable headers extend beyond the borders of the left pane and into the right pane when the page is initially rendered.  If I resize the browser, the TreeTable header changes to the correct size and alignment.  Here's a screen capture when page is initially rendered:

![Capture](https://github.com/primefaces/primefaces/assets/109166456/b9786d05-f658-425a-9ac2-210e65be99b0)
.

### Reproducer

XHTML:

```
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://xmlns.jcp.org/jsf/html"
      xmlns:f="http://xmlns.jcp.org/jsf/core"
      xmlns:p="http://primefaces.org/ui">
    <h:head>
        <title>Title</title>
    </h:head>
    <h:body>
        <h:form id="form">
            <p:splitter gutterSize="10">
                
                <p:splitterPanel>
                    <p:treeTable value="#{ttEventsView.root}" var="document" selectionMode="single"
                                 selection="#{ttEventsView.selectedNode}" size="small"
                                 stickyHeader="true" showGridlines="true">
                        <p:column headerText="Name" sortBy="#{document.name}" filterBy="#{document.name}">
                            <h:outputText value="#{document.name}"/>
                        </p:column>
                        <p:column headerText="Size" sortBy="#{document.size}" filterBy="#{document.size}">
                            <h:outputText value="#{document.size}"/>
                        </p:column>
                        <p:column headerText="Type" sortBy="#{document.type}" filterBy="#{document.type}">
                            <h:outputText value="#{document.type}"/>
                        </p:column>
                    </p:treeTable>
                </p:splitterPanel>
                
                <p:splitterPanel>
                    Row Details go here
                </p:splitterPanel>
                
            </p:splitter>
        </h:form>
    </h:body>
</html>
```

The bean is essentially the same one used in the TreeTable Events showcase (except without the event handlers):

```
@Named("ttEventsView")
@ViewScoped
public class EventsView implements Serializable {

    private TreeNode<Document> root;

    private TreeNode<Document> selectedNode;

    @Inject
    private DocumentService service;

    @PostConstruct
    public void init() {
        root = service.createDocuments();
    }

    public TreeNode<Document> getRoot() {
        return root;
    }

    public TreeNode<Document> getSelectedNode() {
        return selectedNode;
    }

    public void setSelectedNode(TreeNode<Document> selectedNode) {
        this.selectedNode = selectedNode;
    }

    public void setService(DocumentService service) {
        this.service = service;
    }
}
```

### Expected behavior

The TreeTable header row should be correctly aligned when the page is rendered.  Here's what it looks like after the browser is slightly resized:

![Capture2](https://github.com/primefaces/primefaces/assets/109166456/7a94d7b3-bb5f-41f3-be1a-ff272dfcee6b)

### PrimeFaces edition

Community

### PrimeFaces version

13.0.3

### JSF implementation

Mojarra

### JSF version

3.0

### Java version

java-11-openjdk-11.0.21.9-1.windows.redhat.x86_64

**Comment by melloware** [comment]
Please follow the issue template or provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Comment by joepascual** [comment]
https://github.com/joepascual/primefaces-test/tree/treeTable-sticky-header-with-splitter

**Comment by melloware** [comment]
@joepascual here is a working version:
[pf-11055.zip](https://github.com/primefaces/primefaces/files/13504265/pf-11055.zip)

It applies this MonkeyPatch:

```js
if (PrimeFaces.widget.TreeTable) {
    PrimeFaces.widget.TreeTable.prototype_render = function() {
        var $this = this;
        if(this.cfg.scrollable) {
            this.setupScrolling();
        }

        if(this.cfg.filter) {
            this.setupFiltering();
        }

        if(this.cfg.resizableColumns) {
            this.resizableStateHolder = $(this.jqId + '_resizableColumnState');
            this.resizableState = [];

            if(this.resizableStateHolder.attr('value')) {
                this.resizableState = this.resizableStateHolder.val().split(',');
            }

            this.setupResizableColumns();
        }

        if(this.cfg.stickyHeader) {
            setTimeout(function () {$this.setupStickyHeader();}, 1);
        }

        if(this.cfg.editable) {
            this.bindEditEvents();
        }

        this.bindEvents();
    }
};

```

**Comment by joepascual** [comment]
> @joepascual here is a working version: [pf-11055.zip](https://github.com/primefaces/primefaces/files/13504265/pf-11055.zip)
> 
> It applies this MonkeyPatch:
> 
> ```js
> if (PrimeFaces.widget.TreeTable) {
>     PrimeFaces.widget.TreeTable.prototype_render = function() {
>         var $this = this;
>         if(this.cfg.scrollable) {
>             this.setupScrolling();
>         }
> 
>         if(this.cfg.filter) {
>             this.setupFiltering();
>         }
> 
>         if(this.cfg.resizableColumns) {
>             this.resizableStateHolder = $(this.jqId + '_resizableColumnState');
>             this.resizableState = [];
> 
>             if(this.resizableStateHolder.attr('value')) {
>                 this.resizableState = this.resizableStateHolder.val().split(',');
>             }
> 
>             this.setupResizableColumns();
>         }
> 
>         if(this.cfg.stickyHeader) {
>             setTimeout(function () {$this.setupStickyHeader();}, 1);
>         }
> 
>         if(this.cfg.editable) {
>             this.bindEditEvents();
>         }
> 
>         this.bindEvents();
>     }
> };
> ```

Thanks for your help.  I ran the code you provided but I'm still seeing the same rendering bug for the TreeTable's header.

**Comment by melloware** [comment]
can you run my attached zip with `mvn clean jetty:run` ?

**Comment by joepascual** [comment]
> can you run my attached zip with `mvn clean jetty:run` ?

yes

**Comment by melloware** [comment]
Sorry there was a typo:
[pf-11055.zip](https://github.com/primefaces/primefaces/files/13532161/pf-11055.zip)

**Comment by melloware** [comment]
Yeah that might be because once the sticky header is created its not expecting to be resized inside another container.  I bet Datatable has the same issue.

**Comment by joepascual** [comment]
> @joepascual I fixed it again: [pf-11055.zip](https://github.com/primefaces/primefaces/files/13538962/pf-11055.zip)
> 
> Using AJAX `resizeEnd` to redraw the table when the splitter panel changes. that readjusts the table header.
> 
> ```
> <p:splitter gutterSize="10">
> 	 <p:ajax event="resizeEnd" update="myTable" />
>          <p:splitterPanel>    
>                     <p:treeTable id="myTable" 
> ```

Again, thank you for your efforts. I tested the latest fix you provided and I can confirm that the header renders correctly _after_ the splitter is resized.  However, _during active resizing_, the current header remains on the screen (static) while another dynamically resizing header also appears (if that makes sense). I've attached a couple of screenshots while actively resizing the splitter:

![Capture2](https://github.com/primefaces/primefaces/assets/109166456/4dfe39b7-88c9-42fe-97a8-0645f983f574)
![Capture1](https://github.com/primefaces/primefaces/assets/109166456/61f1e53a-fd74-47c2-ab03-a8ebc43072f5)

**Comment by melloware** [comment]
Yep nothing you can do to prevent that.  A sticky header is by default a static size.   This is the best you will get with sticky.

**Comment by joepascual** [comment]
Hopefully this fix will be included in 13.0.4?

**Comment by melloware** [comment]
Yep I already ported it to 13.0.4 which will be released sometime next week.
